crates/types/src/offchain/pool.rs

```diff
@@ -6,7 +7,7 @@ use super::DepositInfo;
 
 pub struct BlockParam {
     pub number: u64,
-    pub block_producer_id: u32,
+    pub block_producer: Bytes,
```

Comment:
You are right, but this value is used to fill the `L2Block#block_producer` field, so if we use `RegistryAddress` we need an extra serialization. IMO we can accept to use `Bytes` here since this structure is only used to generate a block.